aperturedb/ImageDownloader.py

```diff
@@ -173,9 +173,9 @@ def print_stats(self):
         print("Throughput (images/s)):",
               1 / np.mean(times) * self.numthreads)
 
-        print("Total time(s):", self.ingestion_time)
+        print("Total time(s):", self.queries_time)
```

Comment:
This is reading odd, because there are no queries, and there is no Loader :),  Maybe we can define a function called get_aggregated_runtime() in ParallelLoader, and rename the class that Parallelizer. 

Also ParallelQuery sounds to be a specific use case of Parallelizer .
ParallelQuery should not be the base class

Let me know if we should discuss this offline.

Reply:
Ok I gave it try. now we ended up with "batching" notion in Parallelizer, which is not used within that class, but needed in the logic for determining number of workers. I guess it is ok for now, given that the Parallelizer is already implemented.